base_layer/wallet/src/storage/sqlite_db.rs

```diff
@@ -204,31 +230,93 @@ impl WalletSqliteDatabase {
         }
         Ok(())
     }
+
+    fn get_comms_secret_key(&self, conn: &SqliteConnection) -> Result<Option<DbValue>, WalletStorageError> {
+        let cipher = acquire_read_lock!(self.cipher);
+        if let Some(key_str) = WalletSettingSql::get(format!("{}", DbKey::CommsSecretKey), &conn)? {
+            let secret_key = match cipher.as_ref() {
+                None => CommsSecretKey::from_hex(key_str.as_str())?,
+                Some(cipher) => {
+                    let decrypted_key_bytes = decrypt_bytes_integral_nonce(&cipher, from_hex(key_str.as_str())?)
+                        .map_err(|_| WalletStorageError::AeadError("Decryption Error".to_string()))?;
+                    CommsSecretKey::from_bytes(decrypted_key_bytes.as_slice())?
+                },
+            };
+
+            Ok(Some(DbValue::CommsSecretKey(secret_key)))
+        } else {
+            Ok(None)
+        }
+    }
+
+    fn get_comms_address(&self, conn: &SqliteConnection) -> Result<Option<DbValue>, WalletStorageError> {
+        if let Some(key_str) = WalletSettingSql::get(format!("{}", DbKey::CommsAddress), &conn)? {
+            Ok(Some(DbValue::CommsAddress(
+                Multiaddr::from_str(key_str.as_str()).map_err(|_| WalletStorageError::ConversionError)?,
+            )))
+        } else {
+            Ok(None)
+        }
+    }
+
+    fn get_comms_features(&self, conn: &SqliteConnection) -> Result<Option<DbValue>, WalletStorageError> {
+        if let Some(key_str) = WalletSettingSql::get(format!("{}", DbKey::CommsFeatures), &conn)? {
+            let features = u64::from_str(&key_str).map_err(|_| WalletStorageError::ConversionError)?;
+            Ok(Some(DbValue::CommsFeatures(features)))
+        } else {
+            Ok(None)
+        }
+    }
+
+    fn set_tor_id(&self, tor: TorIdentity, conn: &SqliteConnection) -> Result<(), WalletStorageError> {
+        let cipher = acquire_read_lock!(self.cipher);
+        let tor_string = tor.to_json().map_err(|_| WalletStorageError::ConversionError)?;
+        match cipher.as_ref() {
+            None => {
+                WalletSettingSql::new(DbKey::TorID.to_string(), tor_string).set(&conn)?;
+            },
+            Some(cipher) => {
+                let ciphertext_integral_nonce = encrypt_bytes_integral_nonce(&cipher, tor_string.into_bytes())
+                    .map_err(|_| WalletStorageError::AeadError("Encryption Error".to_string()))?;
+                WalletSettingSql::new(DbKey::TorID.to_string(), ciphertext_integral_nonce.to_hex()).set(&conn)?;
+            },
+        }
+
+        Ok(())
+    }
+
+    fn get_tor_id(&self, conn: &SqliteConnection) -> Result<Option<DbValue>, WalletStorageError> {
+        let cipher = acquire_read_lock!(self.cipher);
+        if let Some(key_str) = WalletSettingSql::get(format!("{}", DbKey::TorID), &conn)? {
```

Comment:
nit: `.to_string()`

Reply:
Removed all format!("{}" and replaced with to_string()